### PR TITLE
Add case test_metric_longhorn_backup

### PR DESCRIPTION
**Pull request by roger-ryao**
#### Which issue(s) this PR fixes:
longhorn/longhorn#9430

#### What this PR does / why we need it:
This PR adds a test case to cover the backup metrics.

#### Special notes for your reviewer:
@c3y1huang  @ChanYiLin 

#### Additional documentation or context
https://github.com/longhorn/longhorn/issues/9429

## Summary by CodeRabbit

- **New Features**
	- Introduced enhanced testing for backup metrics, including validation of metrics for user-created and recurring backups.
	- Added a new helper function to improve metric validation across all nodes.
- **Bug Fixes**
	- Improved existing test functions to ensure comprehensive coverage of backup operations.
- **Tests**
	- Added new test functions to validate backup metrics and ensure correct reporting.
	- Defined new constants for recurring job parameters to streamline testing processes.

### CODE REVIEW

File: manager/integration/tests/test_metric.py [hunk 1]
```diff
@@ -615,3 +649,126 @@ def test_node_metrics(client, core_api): # NOQA
...
+                                     user_backup_metric_labels,
+                                     backup_size)
+
+    wait_for_metric_sum_on_all_nodes(client, core_api,
+                                     "longhorn_backup_state",
+                                     user_backup_metric_labels,
+                                     3)
+
+    # delete the existing backup before creating a recurring backup job.
+    delete_backup_volume(client, volume_name)
```
Comment (roger-ryao) [inline manager/integration/tests/test_metric.py:729]: Hi @c3y1huang

In my test case `test_metric_longhorn_backup`, I have referenced `wait_for_metric_count_all_nodes` for similar usage. However, I have a question regarding my test case design. Although I have deleted the backups of volumes that were not created by the recurring job, I am concerned about potential data caching issues when using the same volume to apply a backup recurring job to check the Longhorn backup metric. Could you please help review this test case?

Thanks

Comment (roger-ryao) [reply manager/integration/tests/test_metric.py:729]: After [longhorn/longhorn-manager#3216](https://github.com/longhorn/longhorn-manager/pull/3216) was merged, my test case passed, and I did not observe any potential data caching issues when using the same volume to apply a backup recurring job to check the Longhorn backup metrics.

Screenshot_20241024_154554

**Comment by yangchiu** [comment]
cc @ChanYiLin @c3y1huang as it relates to https://github.com/longhorn/longhorn/issues/9429

**Review by yangchiu — APPROVED** [review]
LGTM

**Pull request merged**